x-pack/plugins/triggers_actions_ui/public/application/sections/alert_form/alert_has_changed.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import deepEqual from 'fast-deep-equal';
+import { pick } from 'lodash';
+import { InitialAlert } from './alert_reducer';
+
+const DEEP_COMPARE_FIELDS = ['tags', 'schedule', 'actions', 'params', 'notifyWhen'];
+
+function getNonNullCompareFields(alert: InitialAlert) {
+  const { name, alertTypeId, throttle } = alert;
+  return {
+    ...(!!(name && name.length > 0) ? { name } : {}),
+    ...(!!(alertTypeId && alertTypeId.length > 0) ? { alertTypeId } : {}),
+    ...(!!(throttle && throttle.length > 0) ? { throttle } : {}),
+  };
+}
```

Comment:
I added tests for the scenarios I saw when manually testing this, namely:
- the `name` field grabs focus when the add alert flyout opens so it starts out as `undefined` but is set to an empty string `onBlur` if the user doesn't enter anything
- the `alertTypeId` field starts out undefined in the stack alerts flyout, but if you select an alert type and then X it out, it is set to null
- the `throttle` field starts out undefined but if you set a throttle and then unset it, it is set to null

This tries to normalize these scenarios so that they can be compared.